WindowManager/app/src/main/java/com/example/windowmanagersample/RxActivity.kt

```diff
@@ -0,0 +1,175 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.windowmanagersample
+
+import android.graphics.drawable.ColorDrawable
+import android.os.Bundle
+import android.util.Log
+import android.view.View
+import androidx.appcompat.app.AppCompatActivity
+import androidx.window.layout.FoldingFeature
+import androidx.window.layout.WindowInfoRepository
+import androidx.window.layout.WindowInfoRepository.Companion.windowInfoRepository
+import androidx.window.layout.WindowLayoutInfo
+import androidx.window.rxjava2.layout.windowLayoutInfoObservable
+import com.example.windowmanagersample.databinding.ActivityRxBinding
+import io.reactivex.Observable
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.Disposable
+import java.text.SimpleDateFormat
+import java.util.Date
+import java.util.Locale
+
+/** Demo activity that shows all display features and current device state on the screen. */
+class RxActivity : AppCompatActivity() {
+
+    private val stateLog: StringBuilder = StringBuilder()
+    private var disposable: Disposable? = null
+
+    private lateinit var binding: ActivityRxBinding
+    private lateinit var windowInfoRepository: WindowInfoRepository
+    private lateinit var observable: Observable<WindowLayoutInfo>
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        binding = ActivityRxBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        windowInfoRepository = windowInfoRepository()
+
+        // Create a new observable
+        observable = windowInfoRepository.windowLayoutInfoObservable()
+
+        stateLog.clear()
+        stateLog.append(getString(R.string.state_update_log)).append("\n")
+    }
+
+    override fun onStart() {
+        super.onStart()
+
+        // Subscribe to receive WindowLayoutInfo updates
+        disposable?.let {
+            if (!it.isDisposed) it.dispose()
+        }
+        disposable = observable
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe() { newLayoutInfo ->
+                updateStateLog(newLayoutInfo)
+                updateCurrentState(newLayoutInfo)
+            }
+    }
+
+    override fun onStop() {
+        super.onStop()
+
+        // Dispose the WindowLayoutInfo observable
+        disposable?.let {
+            if (!it.isDisposed) it.dispose()
+        }
+    }
+
+    /** Updates the device state and display feature positions. */
+    private fun updateCurrentState(layoutInfo: WindowLayoutInfo) {
+        val displayFeatureViews = mutableListOf<View>()
+
+        // Cleanup previously added feature views
+        val rootLayout = binding.featureContainerLayout
+        for (featureView in displayFeatureViews) {
```

Comment:
`displayFeatureViews` will always be empty here (since it was just initialized above), so this `for` loop is essentially a no-op. I think it needs to be defined as before at the `Activity` level, so that it will have the old list of views from before.

Alternatively, we might be able to just use `rootLayout.removeAllViews()`, knowing that we aren't adding anything else to it?

Reply:
I replaced the for loop with `rootLayout.removeAllViews()`

Thanks